backend/src/crd/controller/viewer/reconciler/reconciler.go

```diff
@@ -165,12 +165,16 @@ func setPodSpecForTensorboard(view *viewerV1beta1.Viewer, s *corev1.PodSpec) {
 
 	c := &s.Containers[0]
 	c.Name = view.Name + "-pod"
-	c.Image = "tensorflow/tensorflow"
+	c.Image = view.Spec.TensorboardSpec.TensorflowImage
```

Comment:
Shall we provide a default fallback for old CRDs?

Reply:
If we would like a fallback, we can specify the fallback in FE. E.g., the user input version doesn't parse (when we have this FE input) and FE can hardcode a fallback version. This way we want to check which version is used and why it is used we only need to go to FE code without worrying this version might be later overwritten by a fallback version in BE. 

Moreover, if TB instance fails to be started by controller because of missing or invalid tensorflow image, it might be just ok to let the bug surface and fix it in FE.

Reply:
Considering user who upgrades the cluster. They may have viewer CRs without an image field.

Reply:
reference: https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definition-versioning/

Reply:
> I had a quick fix to unblock the marketplace scenario. #2513
> this proper fix is still needed. please feel free to override the fix i had. thanks!

@IronPan Acked